main.py

```diff
@@ -1,154 +1,210 @@
+            if title == 'Untitled':
+                event_title = '#Warroom'
+            else:
+                event_title = f"#Warroom {title}"
+            headers = {'Authorization': f"Bearer {slack_bot_token}"}
 
+            response = requests.get(url, headers=headers)
+            response.raise_for_status()
+
+            # TODO: this might just need to be response.text
+            content = response.content.decode("utf-8")
+
+            e_time = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(float(event['event_ts'])))
+            e_title = f"{e_time} - {event_title}"
+            username = helper.get_username(event.get('user'), slack_client, slack_bot_token)
+            logger.info(username)
+            logger.info(e_title)
+            logger.info(content)
+            misp_response = misp.misp_send(0, content, e_title, username)
+            slack_client.chat_postEphemeral(
+                channel=event.get('channel'),
+                text=misp_response,
+                user=event.get('user'),
+            )
 
 @slack_events_adapter.on('message')
 def handle_message(event_data):
-	logger.info('Got message from slack')
-	logger.info(event_data)
-	message = event_data['event']
-	if message.get('files') is not None:
-
-		file_info = message
-		t = threading.Thread(target=file_handler, args=(file_info,))
-		t.start()
-		#file_handler(file_info)
-		return flask.Response('', headers={'X-Slack-No-Retry': 1}), 200
-	# if the incoming message contains 'hi', then respond with a 'hello message'
-	if message.get('subtype') is None and 'hi' in message.get('text'):
-		channel = message['channel']
-		message = "Hello <@%s>! :tada:" % message['user']
-		slack_client.chat_postMessage(channel=channel, text=message)
-		return '', 200
+    """ slack message handler """
+    logger.info('Got message from slack')
+    logger.info(event_data)
+    message = event_data.get('event')
+    if message.get('files'):
+        file_info = message
+        thread_object = threading.Thread(target=file_handler, args=(file_info,))
+        thread_object.start()
+        #file_handler(file_info)
+        return_value = flask.Response('', headers={'X-Slack-No-Retry': 1}), 200
+
+    # if the incoming message contains 'hi', then respond with a 'hello message'
+    elif message.get('subtype') is None and 'hi' in message.get('text'):
+        logger.debug(f"Hi message from {message.get('user')} in {message.get('channel')}")
+        response = f"Hello <@{message.get('user')}>! :tada:"
+        slack_client.chat_postMessage(channel=message.get('channel'), text=response)
+        return_value = '', 200
+    # shouldn't get here, but return a 403 if you do.
+    return_value = 'Unhandled message type', 403
+    return return_value
+
 @slack_events_adapter.on("error")
 def error_handler(err):
-	logger.info('got error from slack')
-	print("ERROR: " + str(err))
+    """ slack error message handler """
+    logger.error("Slack error: %s", str(err))
```

Comment:
```suggestion
    logger.error(f"Slack error: {err}")
```

Reply:
Best practice is to use the lazy evaluation as it is.